packages/client/src/runtime/utils/getRuntime.ts

```diff
@@ -1,4 +1,69 @@
-import { detectRuntime, Runtime } from 'detect-runtime'
+// https://runtime-keys.proposal.wintercg.org/
+export type RuntimeName =
+  | 'workerd'
+  | 'deno'
+  | 'netlify'
+  | 'node'
+  | 'bun'
+  | 'edge-light'
+  | '' /* unknown */
+
+  
+function detectRuntime(): RuntimeName {
+  /**
+   * Indicates if running in Node.js or a Node.js compatible runtime.
+   *
+   * **Note:** When running code in Bun and Deno with Node.js compatibility mode, `isNode` flag will be also `true`, indicating running in a Node.js compatible runtime.
+   */
+  const isNode = globalThis.process?.release?.name === 'node'
+  
+  /**
+   * Indicates if running in Bun runtime.
+   */
+  const isBun = !!globalThis.Bun || !!globalThis.process?.versions?.bun
+  
+  /**
+   * Indicates if running in Deno runtime.
+   */
+  const isDeno = !!globalThis.Deno
+  
+  /**
+   * Indicates if running in Netlify runtime.
+   */
+  const isNetlify = typeof globalThis.Netlify === 'object'
+  
+  /**
+   * Indicates if running in EdgeLight (Vercel Edge) runtime.
+   */
+  const isEdgeLight = typeof globalThis.EdgeRuntime === 'object'
```

Comment:
Why different styles of checks (i.e. `!!globalThis.X` vs `typeof globalThis.X === 'object')?

Reply:
Because the second is more accurate, and was originally used in `detect-runtime`